src/org/chartjs/CandlestickDAOChartCView.js

```diff
@@ -0,0 +1,204 @@
+        }
+      }));
+    },
+
+    function generateConfig() {
+      var config = {
+        type: this.chartType,
+        data: this.generateData(),
+        options: this.generateOptions()
+      };
+
+      return config;
+    },
+
+    function generateData() {
+      var configData = {
+        datasets: this.generateDataSets()
+      };
+
+      return configData;
+    },
+
+    function generateDataSets() {
+      var self = this;
+      var datasets = [];
+      // Each dataset is represented by a key in the CandlestickDAO
+      Object.keys(this.candlestickMap).forEach(function( key ) {
+        const candlesticks = self.candlestickMap[key].array;
+        var dataset = {};
+
+        // Array of points that will represent all the candlesticks that share the same key
+        var pointData = [];
+        candlesticks.forEach(function ( candlestick ) {
+          var point = {};
+          point['x'] = candlestick.closeTime;
+          point['y'] = candlestick[self.dataPointProperty.name];
+          pointData.push(point);
+        });
+        dataset['data'] = pointData;
+
+        // Default title to represent the dataset
+        dataset['label'] = key;
+
+        // This should allow maximum configurability by devs.
+        // Refer to chartjs.org documentation for dataset properties
+        if ( self.customDatasetStyling ) {
+          // If custom styling is provided for any key
+          var customStyling = self.customDatasetStyling[key];
+          if ( customStyling ) {
+            // If custom styling is provided for this specific key
+            Object.keys(customStyling).forEach(function( key ) {
+              dataset[key] = customStyling[key];
+            });
+          }
+        }
+
+        datasets.push(dataset);
+      });
+
+      return datasets;
+    },
+
+    function generateOptions() {
+      var options = {};
+
+      // Default X-Axis scale for candlesticks.
+      options['scales'] = {
+        xAxes: [{
+          type: 'time',
+          distribution: 'linear'
+        }]
+      };
+
+      // This should allow maximum configurability by devs.
+      // Refer to chartjs.org documentation for dataset properties
+      if ( this.customChartOptions ) {
+        var self = this;
+        Object.keys(this.customChartOptions).forEach(function( key ) {
+          options[key] = self.customChartOptions[key];
+        });
+      }
+
+      return options;
+    }
+  ],
+
+  listeners: [
+    {
+      name: 'dataUpdate',
+      isFramed: true,
+      code: function() {
+        // Data (which should be a CandlestickDAO) has been updated
+        if ( ! this.data ) {
+          // data was set to null
+          return;
+        }
+
+        var self = this;
+        // Get all candlesticks categorized by their key, and ordered in ascending order by their closeTime.
+        this.data.orderBy(this.Candlestick.CLOSE_TIME).select(this.GROUP_BY(this.Candlestick.KEY, this.ArraySink.create())).then( function(a) {
```

Comment:
There's a bunch of jumping around here that makes this kinda tough to read.
generateConfig goes to generateData which goes to generateDataSets, etc.

Check out https://github.com/foam-framework/foam2/pull/2103 and the test that's in the comment. I think using that approach to get the dataset together can help clean this up a bunch and I think the entire config can be done in one method since I don't see us needing to reuse much of these methods.

Reply:
I personally find that to be pretty hard to read. I'll see what I can do simplify the functions